### DISCUSSION TITLE
Given an attachment key, how can I find the (parent) item key?

### DISCUSSION THREAD

**Discussion by mhucka** [retorquere/zotero-better-bibtex#1953] [General]
I have successfully used the incredibly useful BBT [json-rpc](https://retorque.re/zotero-better-bibtex/exporting/json-rpc/) interface for a number of operations, but cannot figure out how to do the following: given an attachment key, can I find out the item key for the Zotero record that contains that attachment?

More specifically, following the [naming scheme used by the Zotero devs](https://forums.zotero.org/discussion/comment/335046/#Comment_335046), if I'm looking at a PDF file on disk in my Zotero storage area, and this file has a path of the form `..../storage/PPMJQGRI/file.pdf`, the folder name `PPMJQGRI` is the "attachment key". Using [Pyzotero](https://pyzotero.readthedocs.io/en/latest/), it's possible to look up the parent record key, [which I do](https://github.com/mhucka/zowie/blob/0603a6582bdca9c9e73f5b90b7dc916f758fd669/zowie/zotero.py#L132-L170) in a program I wrote called [Zowie](https://github.com/mhucka/zowie). Now I'm trying to see if I can use BBT to get this information instead of using Pyzotero.

It seems like BBT has everything needed, but I can't find a suitable rpc endpoint to make it work. The closest I can come is to export the entire database in (say) "jzon" format and then look for the file in the result, but this is much too slow.  Is there a more direct way?

**Comment by mhucka** [comment]
<s>Update: I tried one more thing, and it actually worked for a limited case: if I use the `item.search` json-rpc endpoint, and use the attachment key (i.e., folder name) as the search term, BBT returns the record _if_ the attachment key is something in my personal library and not a group library. This is half-way there and I'm happy to have found this much. So my question is now reduced to the following: can I search with an attachment key in shared/group databases too?</s> Update: no, that's not true. I must have accidentally used an item key instead of an attachment key when this happened. Ignore this comment.

My original question stands as-is.

**Comment by retorquere** [comment]
No problem. If you open an issue, I'll get you an endpoint; it needs to be on an issue as I can only drop test builds there.
